File: macroslib/tests/expectations/interface_with_str.rs
foreign_callback!(callback SomeObserver {
    self_type SomeTrait;
    onStateChanged = SomeTrait::on_state_changed(&self, _: &str);
});

foreigner_class!(class ClassWithCallbacks {
    self_type Foo;
    constructor Foo::default() -> Foo;
    method f1(&mut self, cb: Box<dyn SomeTrait>);
});

File: macroslib/tests/expectations/interface_with_str.java
r#"public final void f1(@NonNull SomeObserver cb) {
        do_f1(mNativeObj, cb);
    }
    private static native void do_f1(long self, SomeObserver cb);"#;

r#"public interface SomeObserver {


    void onStateChanged(@NonNull String a0);

}"#;
